src/unix/encoding.ml

```diff
@@ -0,0 +1,201 @@
+(* This file is part of Dream, released under the MIT license. See LICENSE.md
```

Comment:
Why is this file in `src/unix/`? Can't we do compression/decompression "purely," i.e. in memory?